src/elements/Svg.tsx

```diff
@@ -125,6 +124,31 @@ export default class Svg extends Shape<SvgProps> {
       strokeLinejoin,
       strokeMiterlimit,
     } = stylesAndProps;
+
+    // Ensure width and height are treated correctly if they are percentages
+    const isWidthPercentage = typeof width === 'string' && width.includes('%');
+    const isHeightPercentage =
+      typeof height === 'string' && height.includes('%');
+
+    if (viewBox) {
+      const viewBoxValues = viewBox.split(' ');
+      if (viewBoxValues.length === 4) {
+        const viewBoxWidth = parseFloat(viewBoxValues[2]);
+        const viewBoxHeight = parseFloat(viewBoxValues[3]);
+        if (!isNaN(viewBoxWidth) && !isNaN(viewBoxHeight)) {
+          if (width !== undefined && height === undefined) {
+            height = isWidthPercentage
+              ? `${(parseFloat(width) * viewBoxHeight) / viewBoxWidth}%`
```

Comment:
I dont think that's how `%` value should be calculated. For example, let's take that screen width is `200` and height is `600`. Now we have SVG with VB proportion `1` (like `100/100`) and width `50%`. Height is expected to be the same as width while in this implementation it will end up as `50%` of the height which is `300`, not `100`

I don't think the `%` value is being calculated correctly. For instance, if the screen width is `200` and the height is `600`, and we have an SVG with a viewBox proportion of `1` (such as `100/100`) and a width set to `50%`, the height should match the width. However, in this implementation, the height ends up being `50%` of the screen height, which is `300`, rather than the expected `100`.